### PR TITLE
Intermediate changes to enable HS to monitor event's metadata

### PR DESCRIPTION
**Pull request by menanteau**
Translating strings to enums and back.

Remove test function

### CODE REVIEW

**Review by srp3rd — APPROVED** [review]
Everything looks fine

File: python/HeaderService/hslib_salobj.py [hunk 1]
```diff
@@ -1074,6 +1127,62 @@ def collect_from_HeaderService(self, imageName):
...
+                isenum = True
+                latest_value = metadata[keyword]
+                current_value = self.metadata[imageName][keyword]
+                device = self.config.telemetry[keyword]['device']
+                array_name = self.config.telemetry[keyword]['array_name']
+                device_lib = getattr(self.idl_lib[device], array_name)
+                latest_value = getattr(device_lib, latest_value).numerator
+                current_value = getattr(device_lib, current_value).numerator
+            else:
+                msg = f"arraty: {self.config.telemetry[keyword]['array']} not supported for monitor"
```
Comment (tribeiro) [inline python/HeaderService/hslib_salobj.py]: typo? arraty -> array

**Review by tribeiro — APPROVED** [review]
Looks great! thanks for the updates!

I think I found one typo though.

File: python/HeaderService/hslib_salobj.py (hunk 1 above)
Comment (menanteau) [reply python/HeaderService/hslib_salobj.py]: Thank you. Fixed.

**Pull request merged**